src/main/java/com/_up/megastore/controllers/interfaces/IProductController.java

```diff
@@ -18,4 +21,16 @@ ProductResponse saveProduct(
             @RequestPart MultipartFile multipartFile
     );
 
+    @GetMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    ProductResponse findProductById(@PathVariable UUID id);
+
+    @GetMapping()
+    @ResponseStatus(HttpStatus.OK)
+    Page<ProductResponse> findProductsByPages(
+            @RequestParam(defaultValue = "0") int page,
+            @RequestParam(defaultValue = "15") int pageSize,
+            @RequestParam(defaultValue = "name") String sortBy,
+            @RequestParam(defaultValue = "") String filter
```

Comment:
Siguiendo el comentario anterior, aca deberia cambiarse a name

```suggestion
            @RequestParam(defaultValue = "") String name
```